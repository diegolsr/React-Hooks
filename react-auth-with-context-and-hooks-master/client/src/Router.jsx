import React, { useEffect, useContext } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import { UserContext } from './context/userContext';
import { authenticate, signup, login } from './actions/actions';
import StyledContainer from './components/StyledContainer';
import NavBar from './components/NavBar';
import Login from './components/LoginForm';
import Signup from './components/SingupForm';

export default function Router() {
	const [userState, userDispatch] = useContext(UserContext);
	useEffect(() => {
		async function fetchUser() {
			const userDoc = await authenticate();
			if (userDoc) {
				userDispatch({
					type: 'SET_USER',
					payload: userDoc
				});
			}
			return [];
		}
		fetchUser();
	}, [userDispatch]);
	return (
		<BrowserRouter>
			<NavBar user={userState.user !== null} />
			{userState.user ? <UserView /> : <PublicView />}
		</BrowserRouter>
	);
}

const UserView = () => (
	<Switch>
		<Route
			exact
			path="/"
			render={props => (
				<StyledContainer {...props} title="User Home">
					<div>User Home Placeholder</div>
				</StyledContainer>
			)}
		/>
		<Route path="*" render={() => <Redirect to="/" />} />
	</Switch>
);

const PublicView = () => (
	<Switch>
		<Route
			exact
			path="/"
			render={props => (
				<StyledContainer {...props} title="Public Home">
					<div>Public Home Placeholder</div>
				</StyledContainer>
			)}
		/>
		<Route
			exact
			path="/login"
			render={props => (
				<StyledContainer {...props} title="Login">
					<Login handleLogin={loginData => login(loginData)} />
				</StyledContainer>
			)}
		/>
		<Route
			exact
			path="/signup"
			render={props => (
				<StyledContainer {...props} title="Signup">
					<Signup handleSignup={signupData => signup(signupData)} />
				</StyledContainer>
			)}
		/>
		<Route path="*" render={() => <Redirect to="/" />} />
	</Switch>
);
