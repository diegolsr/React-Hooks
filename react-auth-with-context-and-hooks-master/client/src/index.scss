//imports and variables
@charset "utf-8";
@import "../node_modules/bulma/bulma.sass";
@import url("https://use.fontawesome.com/releases/v5.8.1/css/all.css")