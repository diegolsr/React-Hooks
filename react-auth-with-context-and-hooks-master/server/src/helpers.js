import bcrypt from 'bcrypt';
import User from '../models/user';
import { logger } from '../configs/loggerConfig';

const encrypt = password => bcrypt.hashSync(password, bcrypt.genSaltSync(10));

const sanitize = userDoc => {
	userDoc.password = undefined;
	userDoc.__v = undefined;
	return userDoc;
};

export const send = (userDoc, res) => res.status(200).json(sanitize(userDoc));

export const findByEmail = email =>
	User.findOne({ email: email })
		.then(user => user)
		.catch(err => {
			logger.log(err);
			return err;
		});

export const findByUsername = username =>
	User.findOne({ username: username })
		.then(user => user)
		.catch(err => {
			logger.log(err);
			return err;
		});

export const createUser = (username, password, email) =>
	User.create({
		username: username,
		password: encrypt(password),
		email: email
	})
		.then(user => user)
		.catch(err => {
			logger.log(err);
			return err;
		});
