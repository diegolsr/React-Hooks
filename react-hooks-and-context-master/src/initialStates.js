export const initialTodos = [
	{
		text: 'Eat Dinner',
		completed: false
	}
];

export const initialErrors = {
	header: '',
	form: ''
};
