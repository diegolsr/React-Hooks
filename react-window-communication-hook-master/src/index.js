const { useState, useEffect, useRef } = require('react');

const initialState = {
  lastMessage: undefined,
  messages: []
};

const supportsBroadcastAPI = (() => window && window.BroadcastChannel)();

function useBrowserContextCommunication(channelName) {
  if (channelName === undefined) {
    throw Error('You need to pass a channel name e.g. useBrowserContextCommunication("GreatChannel")');
  }


  const [ state, setMessages ] = useState(initialState);
  let channel;

  if (supportsBroadcastAPI) {
    channel = useRef(new BroadcastChannel(channelName));
  }

  function postMessage(message) {
    if (message) {
      const msg = JSON.stringify({
        message,
        time: Date.now()
      });

      if (supportsBroadcastAPI && channel) {
        channel.current.postMessage(msg);
      } else {
        window.localStorage.setItem(channelName, msg);
      }
    }
  }

  function updateState(data) {
    setMessages(prevState => {
      return {
        lastMessage: data.message,
        messages: prevState.messages.concat(data.message)
      };
    });
  }

  function updateFromLocalStorage(e) {
    const data = JSON.parse(e.newValue);
    updateState(data);
  }

  useEffect(
    () => {
      if (supportsBroadcastAPI) {
        if (channel && channel.current) {
          channel.current.onmessage = e => updateState(JSON.parse(e.data));
        }
      } else {
        window.addEventListener('storage', updateFromLocalStorage);
      }

      return function cleanup() {
        if (channel.current) {
          channel.current.close();
          channel.current = null;
        } else {
          window.localStorage.removeItem(channelName);
          window.removeEventListener('storage', updateFromLocalStorage);
        }
      };
    },
    [ channelName ]
  );

  return [ state, postMessage ];
}

module.exports = useBrowserContextCommunication;
