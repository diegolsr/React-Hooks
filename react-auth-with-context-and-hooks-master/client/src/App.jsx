import React from "react";
import Router from "./Router";
import ContextProvider from "./context/contextProvider";
import ErrorBoundary from "./errorBoundary";

export default function App(props) {
  return (
    <ErrorBoundary>
      <ContextProvider>
        <div className="container">
          <Router />
        </div>
      </ContextProvider>
    </ErrorBoundary>
  );
}
