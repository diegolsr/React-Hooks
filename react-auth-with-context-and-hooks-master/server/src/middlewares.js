import passport from 'passport';
import { createUser, findByEmail, findByUsername, send } from './helpers';
import {
	emailInUse,
	missingData,
	unauthorized,
	usernameInUse
} from '../configs/errorsConfig';

export const check = async (req, res, next) => {
	const { type, value } = req.body;
	try {
		const result =
			type === 'username'
				? await findByUsername(value)
				: await findByEmail(value);
		if (!result) {
			res.status(200).json({
				true: 1
			});
		} else
			res.status(200).json({
				true: 0
			});
	} catch (err) {
		next(err);
	}
};

export const authenticate = (req, res, next) => {
	try {
		req.isAuthenticated() ? send(req.user, res) : next(unauthorized);
	} catch (err) {
		next(err);
	}
};

export const login = (req, res, next) =>
	passport.authenticate('local', (err, user) =>
		err
			? next(err)
			: !user
				? next(unauthorized)
				: req.login(user, err => (err ? next(err) : send(req.user, res)))
	)(req, res, next);

export const signup = async (req, res, next) => {
	const { username, email, password } = req.body;
	if (!email || !password || !username) next(missingData);
	else
		try {
			let foundUser = await findByEmail(email);
			if (foundUser) next(emailInUse);
			else {
				foundUser = await findByUsername(username);
				if (foundUser) next(usernameInUse);
				else {
					const createdUser = await createUser(
						email,
						password,
						username
					);
					req.login(createdUser, err =>
						err ? next(err) : send(createdUser, res)
					);
				}
			}
		} catch (err) {
			next(err);
		}
};
