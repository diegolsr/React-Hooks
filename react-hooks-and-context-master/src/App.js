import React, { useReducer } from 'react';
import { initialTodos, initialErrors } from './initialStates';
import Header from './components/Header';
import AddToDoForm from './components/AddToDoForm';
import ToDoList from './components/ToDoList';
import ErrorButtons from './components/ErrorButtons';
import todoReducer from './reducers/todoReducer';
import errorReducer from './reducers/errorReducer';
import './App.css';

export const TodosDispatch = React.createContext(null);
export const ErrorsContext = React.createContext(null);

const App = () => {
	const [ todos, todosDispatch ] = useReducer(todoReducer, initialTodos);
	const [ errors, errorDispatch ] = useReducer(errorReducer, initialErrors);

	return (
		<div className="App">
			<TodosDispatch.Provider value={todosDispatch}>
				<ErrorsContext.Provider value={[ errors, errorDispatch ]}>
					<Header />
					<AddToDoForm />
				</ErrorsContext.Provider>
				<ToDoList todos={todos} />
				<ErrorsContext.Provider value={errorDispatch}>
					<ErrorButtons />
				</ErrorsContext.Provider>
			</TodosDispatch.Provider>
		</div>
	);
};

export default App;
