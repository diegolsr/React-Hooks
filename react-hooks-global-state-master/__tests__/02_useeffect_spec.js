import React, { useEffect } from 'react';
import { render, cleanup } from 'react-testing-library';

import { createGlobalState } from '../src/index';

describe('useeffect spec', () => {
  afterEach(cleanup);

  it('should update a global state with useEffect', () => {
    const initialState = {
      counter1: 0,
    };
    const { GlobalStateProvider, useGlobalState } = createGlobalState(initialState);
    const Counter = () => {
      const [value, update] = useGlobalState('counter1');
      useEffect(() => {
        update(9);
      }, [update]);
      return (
        <div>
          <span>{value}</span>
        </div>
      );
    };
    const App = () => (
      <GlobalStateProvider>
        <Counter />
      </GlobalStateProvider>
    );
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
