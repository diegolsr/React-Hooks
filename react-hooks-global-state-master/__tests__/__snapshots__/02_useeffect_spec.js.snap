// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useeffect spec should update a global state with useEffect 1`] = `
<div>
  <div>
    <span>
      9
    </span>
  </div>
</div>
`;
